python/ray/dataframe/dataframe.py

```diff
@@ -1072,9 +1072,16 @@ def as_blocks(self, copy=True):
             "github.com/ray-project/ray.")
 
     def as_matrix(self, columns=None):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        """Convert the frame to its Numpy-array representation.
+
+        Args:
+            columns: If None, return all columns, otherwise,
+                returns specified columns.
+
+        Returns:
+            values: ndarray
+        """
+        return to_pandas(self).as_matrix(columns)
```

Comment:
`__array__` does the same thing here. Would be better if this called that under the hood, so that it can be optimized in one place later.

Reply:
do you think b/c `as_matrix` has the `columns` kwargs, that we should leave `as_matrix` like it is now, but then use `return self.as_matrix()` for `__array__`, so it will be optimized in one place, but we can deal with the columns?

Reply:
Sure, that works too.

Reply:
After review, it looks like they do similar things, but `__array__` takes `dtypes` (our implementation is just disregarding that for some reason), and `as_matrix` takes columns, so let's just keep them separate for now. I'll add a TODO here also, though.